craft_application/launchpad/models/recipe.py

```diff
@@ -0,0 +1,351 @@
+#  This file is part of craft-application.
+#
+#  Copyright 2024 Canonical Ltd.
+#
+#  This program is free software: you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License version 3, as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+#  SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Recipe classes."""
+
+# This file relies heavily on dynamic features from launchpadlib that cause pyright
+# to complain a lot. As such, we're disabling several pyright checkers for this file
+# since in this case they generate more noise than utility.
+# pyright: reportUnknownMemberType=false
+# pyright: reportUnknownVariableType=false
+# pyright: reportUnknownArgumentType=false
+# pyright: reportOptionalMemberAccess=false
+# pyright: reportAttributeAccessIssue=false
+# pyright: reportOptionalCall=false
+# pyright: reportOptionalIterable=false
+
+from __future__ import annotations
+
+import enum
+import time
+import typing
+from collections.abc import Collection, Iterable
+from typing import TYPE_CHECKING
+
+import lazr.restfulclient.errors  # type: ignore[import-untyped]
+from typing_extensions import Any, Self, TypedDict, override
+
+from .. import errors, util
+from . import build
+from .base import LaunchpadObject, Pocket
+
+if TYPE_CHECKING:
+    from .. import Launchpad
+
+
+class RecipeType(enum.Enum):
+    """The type of recipe."""
+
+    SNAP = "snap"
+    CHARM = "charm_recipe"
+
+
+class BuildChannels(TypedDict, total=False):
+    """Typed dictionary for the snap channels for recipe builds."""
+
+    # Snaps and charms
+    core: str
+    core18: str
+    core20: str
+    core22: str
+    core24: str
+
+    charmcraft: str  # Charms only
+
+    # Snaps only
+    snapcraft: str
+    snapd: str
+
+
+class _BaseRecipe(LaunchpadObject):
+    name: str
+    """The name of the recipe."""
+    owner_name: str
+    """The name of the owner."""
+
+    _resource_types = RecipeType
+    _attr_map = {"owner_name": "owner.name"}
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}.get(name={self.name!r}, owner={self.owner_name!r})"
```

Comment:
I am not sure if this is suppose to be formatted to`"NAME.get(name=N, owner=O)"` or you want to call `NAME.get(name=N, owner=O)` and get the result.

Reply:
It's a bit weird and I'll add a comment, but I'm intending to format it the first way.